drivers/spi/spi_psoc6.c

```diff
@@ -0,0 +1,419 @@
+/*
+ * Copyright (c) 2021 ATL Electronics.
```

Comment:
This file appears to be based on `drivers/spi/spi_mcux_dspi.c`. Please carry over the copyright

Reply:
Indeed, some ideas came from `spi_mcux_dspi.c`. Added credits as requested.